src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -268,21 +251,14 @@ public void removeCitizen(final CitizenData citizen)
     }
 
     /**
-     * Get the max tool level useable by the worker.
+     * Returns if the {@link CitizenData} is the same as the worker.
      *
-     * @return the integer.
+     * @param citizen {@link CitizenData} you want to compare
+     * @return true if same citizen, otherwise false
      */
-    public int getMaxToolLevel()
+    public boolean isWorker(final CitizenData citizen)
     {
-        if (getBuildingLevel() >= getMaxBuildingLevel())
-        {
-            return TOOL_LEVEL_MAXIMUM;
-        }
-        else if (getBuildingLevel() <= WOOD_HUT_LEVEL)
-        {
-            return TOOL_LEVEL_WOOD_OR_GOLD;
-        }
-        return getBuildingLevel() - WOOD_HUT_LEVEL;
+        return workers.contains(citizen);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2250)